src/typecheck.cpp

```diff
@@ -40,7 +40,7 @@ void typecheck(TypedAST::ArrayLiteral* ast, TypeChecker& tc) {
 	auto element_type = tc.new_var();
 	for (auto& element : ast->m_elements) {
 		typecheck(element.get(), tc);
-		tc.m_core.unify(element_type, element->m_value_type);
+		tc.m_core.m_mono_core.unify(element_type, element->m_value_type);
```

Comment:
I don't know if we should put the various unifiers directly in the `TypeChecker`?

Reply:
I literally don't know

Reply:
I considered this. I'm not sure either.

Pros:
- more abstraction possibly?
- less accessing through the various components

Cons:
- name clashing (3 billion of each function)

Name clashing is something that triggers me, because I have to rename my functions... It's minor but that's why I left it like that. What do you think?

Reply:
what? the member variables are already called `m_mono_something` what would they collide with? what functions are you talking about?

Reply:
Oh I think I understood wrong, you are saying that we put the Unification::Cores in the type checker. Also because there would be so little left in the type system core we could get rid of it altogether?

Reply:
yup, that is what I meant. Yeah, we can look at this later.